### ISSUE TITLE
InstanceName is used finding the factory for a dependency

### ISSUE THREAD

**Issue by n3wtron** [fluttercommunity/get_it#179]
Hi,
I think that I've found a bug registering a SingletonWithDependencies and using an instanceName.

```
GetIt.I.registerSingletonWithDependencies<MyClass>(
  () => MyClass(),
  instanceName: "myInstanceName",
  dependsOn : [OtherClass]
);
```

in `get_it_impl.dart` line (789)
```
for (final type in dependsOn!) {
  final dependentFactory =
-->>      _findFirstFactoryByNameAndTypeOrNull(instanceName, type);    <---
```
is looking for the Factory of the `OtherClass` but is using the current(MyClass) instanceName.

I don't think is the correct behavior, the `OtherClass` can have a different instanceName or not have it at all.

**Possible quick fix**
_I'm pretty new with dart, so take this fix as a newbie suggestion_
 change this
` _findFirstFactoryByNameAndTypeOrNull(instanceName, type);`
with
` _findFirstFactoryByNameAndTypeOrNull(null, type);`

a more complete fix should involve the `dependsOn` array that should allow to put type+possibleInstanceName

**Comment by escamoteur** [comment]
Yep, I think you spotted it correctly and indeed `null` should be passed there. As I never used instanceNames in my own project they don't get as much tested unfortunately

**Comment by escamoteur** [comment]
Included in V7.0.0
